core/src/main/java/org/bitcoinj/base/internal/ByteUtils.java

```diff
@@ -293,32 +382,42 @@ public static long readInt64(byte[] bytes, int offset) throws ArrayIndexOutOfBou
      *                                        if the read value extends beyond its bounds
      */
     public static long readUint32BE(byte[] bytes, int offset) throws ArrayIndexOutOfBoundsException {
-        return ((bytes[offset] & 0xffL) << 24) |
-                ((bytes[offset + 1] & 0xffL) << 16) |
-                ((bytes[offset + 2] & 0xffL) << 8) |
-                (bytes[offset + 3] & 0xffL);
+        check(offset >= 0 && offset <= bytes.length - 4, () ->
+                new ArrayIndexOutOfBoundsException(offset));
+        return readUint32BE(ByteBuffer.wrap(bytes, offset, bytes.length - offset));
     }
 
     /**
-     * Read 2 bytes from the byte array (starting at the offset) as unsigned 16-bit integer in big endian format.
+     * Read 8 bytes from the buffer as signed 64-bit integer in little endian format.
+     * @param buf buffer to be read from
+     * @throws BufferUnderflowException if there are fewer bytes than needed available in the buffer
+     */
+    public static long readInt64(ByteBuffer buf) throws BufferUnderflowException {
+        return buf.order(ByteOrder.LITTLE_ENDIAN).getLong();
+    }
+
+    /**
+     * Read 8 bytes from the byte array (starting at the offset) as signed 64-bit integer in little endian format.
      * @param bytes buffer to be read from
      * @param offset offset into the buffer
      * @throws ArrayIndexOutOfBoundsException if offset points outside of the buffer, or
      *                                        if the read value extends beyond its bounds
      */
-    public static int readUint16BE(byte[] bytes, int offset) throws ArrayIndexOutOfBoundsException {
-        return ((bytes[offset] & 0xff) << 8) |
-                (bytes[offset + 1] & 0xff);
+    public static long readInt64(byte[] bytes, int offset) throws ArrayIndexOutOfBoundsException {
+        check(offset >= 0 && offset <= bytes.length - 8, () ->
+                new ArrayIndexOutOfBoundsException(offset));
+        return readInt64(ByteBuffer.wrap(bytes, offset, bytes.length - offset));
     }
 
     /**
      * Read 2 bytes from the stream as unsigned 16-bit integer in little endian format.
      * @param is stream to be read from
      */
     public static int readUint16FromStream(InputStream is) {
```

Comment:
I suppose this could be renamed to just `readUnit16` (in a separate PR)

Reply:
Yes, agree.

Reply:
Renamed in 7303043f37a0a073841a7a368e3fb2a360b1a7b8.